### PR TITLE
Generate stable test output

### PR DESCRIPTION
**Pull request by florentx**
Currently the output of the tests is not stable when running in different environments:
- some absolute file paths are printed, which depend on the path of the build directory
- `Last updated` timestamp is printed in the output of many tests

This PR removes the above glitches. It is a prerequisite for fixing issue #30

Details of the changes:
- florentx@1c49ffe patches the `time.localtime` function and the `TZ` environment variable which are used to generate the `Last updated` timestamp
- florentx@b3fb9cc adds a stub `barchart.py` file to avoid printing an error message with the absolute path of the missing file
- florentx@f4c5f46 replaces absolute path `{docfile}` with relative path `{docname}.txt`

### CODE REVIEW

**Comment by florentx** [comment]
Additional fixes to avoid depending on version numbers for the expected output:
- florentx@ebc4078 changes output:

``` diff
-<div class="content"><!-- Generator: GNU source-highlight 3.1.7
+<div class="content"><!-- Generator: GNU source-highlight
```
- florentx@2e0bc77 changes output:

``` diff
-<meta name="generator" content="AsciiDoc 8.6.9">
+<meta name="generator" content="AsciiDoc test-0.1.1">
```

File: tests/testasciidoc.py [hunk 1]
```diff
@@ -75,6 +76,28 @@ def normalize_data(lines):
     return result
 
 
+def mock_localtime(f, _localtime=time.localtime):
+    """Mock time module to generate stable output."""
+    _frozentime = 0X3DE170D6
+    _frozentz = 'Pacific/Auckland'
+
+    def _fake_localtime(t=_frozentime + 1):
```
Comment (florentx) [inline tests/testasciidoc.py]: could be named `_frozen_localtime` instead of `_fake_localtime` to be consistent.

File: tests/testasciidoc.py [hunk 2]
```diff
@@ -84,7 +107,7 @@ def __init__(self):
         self.description = []   # List of lines followoing title.
         self.source = None      # AsciiDoc test source file name.
         self.options = []
-        self.attributes = {}
+        self.attributes = {'asciidoc-version': 'test-%s' % __version__}
```
Comment (florentx) [inline tests/testasciidoc.py]: a safer alternative is to ignore this `__version__` too, and expect output:

``` html
<meta name="generator" content="AsciiDoc test">
```

**Comment by michel-kraemer** [comment]
+1

What about your comments? Are you still working on this PR or should we merge it?

Cheers,
Michel

**Comment by florentx** [comment]
I think I will apply the 2 suggestions I've proposed in comments : re-reading them after some days, I am +1 with myself :smile: 

Then this PR is ready for merging, and I could work to finish #30.

**Comment by aerostitch** [comment]
Hi,

For the Last updated field, you could also use the footer-style attribute to none as default in the tests. This would remove the last updated field in the html pages for the tests.
This attribute is introduced by #9 .

Joseph

File: filters/source/source-highlight-filter.conf [hunk 3]
```diff
@@ -58,7 +58,7 @@ endif::basebackend-docbook[]
 ifdef::basebackend-html[]
 [source-filter-style]
 ifeval::["{source-highlighter}"=="source-highlight"]
-source-style=template="source-highlight-block",presubs=(),postsubs=("callouts",),posattrs=("style","language","src_numbered","src_tab"),filter="source-highlight -f xhtml -s {language} {src_numbered?--line-number=' '} {src_tab?--tab={src_tab}} {args=}"
+source-style=template="source-highlight-block",presubs=(),postsubs=("callouts",),posattrs=("style","language","src_numbered","src_tab"),filter="source-highlight --gen-version -f xhtml -s {language} {src_numbered?--line-number=' '} {src_tab?--tab={src_tab}} {args=}"
 endif::[]
```
Comment (elextr) [inline filters/source/source-highlight-filter.conf:62]: why does adding the generated version help to produce stable output?

Comment (florentx) [inline filters/source/source-highlight-filter.conf:62]: actually the switch `--gen-version` toggle off the version number

Comment (elextr) [inline filters/source/source-highlight-filter.conf:62]: Ok, makes sense (well your change does, the option doesn't, should be --no-gen-version).

**Comment by elextr** [comment]
Are you able to squash the commits correcting previous commits? Leaving 4 commits.

**Comment by florentx** [comment]
ok, I'll try to rewrite the history

**Comment by florentx** [comment]
I've rebased and merged the commits.
Build passes.

**Pull request merged**